pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -162,8 +162,9 @@ def on_train_batch_start(
             # First log at beginning of second step
             logs["batch_time/inter_step (ms)"] = (time.time() - self._snap_inter_step_time) * 1000
 
-        assert trainer.logger is not None
-        trainer.logger.log_metrics(logs, step=trainer.global_step)
+        assert trainer.loggers
```

Comment:
True, it's not required, but do we want to keep it for debugging purposes? Rather than silently not logging anything?

Reply:
I would assume debugging was the original intention, because otherwise the line would be `if trainer.logger is not None` rather than `assert trainer.logger is not None`, in which case it would still makes sense to keep this.

Reply:
Any assertions inside the code (not tests) are there just to notify mypy that we don't expect that the value is `None`, never for debugging

Reply:
> Any assertions inside the code (not tests) are there just to notify mypy that we don't expect that the value is `None`, never for debugging

Oh that's good to know, thank you. There are some other places in the code where mypy was giving me issues so this will be useful.